src/dataMigration.js

```diff
@@ -239,6 +239,24 @@ const dataMigrations = [
       });
     },
   },
+
+  // 5.1.2 fixes a sync out queue bug which was preventing invalid sync out records being
+  // overwritten. On upgrade, the following migration code filters the sync queue for invalid
+  // sync out records and deletes any which are found.
+  {
+    version: '5.1.2',
+    migrate: database => {
+      database.write(() => {
+        const syncRecords = database.objects('SyncOut');
+        const invalidSyncRecords = syncRecords.filter(syncRecord => {
```

Comment:
do we need to `slice()` to turn this into an array?

Reply:
I believe `Realm.Results` inherits/implements `array.prototype.filter` now (not sure how recently), and behaves similarly to `array.prototype.map` (i.e. implicitly coerces to `array`). I also wasn't sure so cross-referenced with other parts of our codebase, and we do have it a few places, e.g. (from `PeriodSchedule.js`):

```
return isEmergency ? this.periods.slice() : this.periods.filter(filterValidPeriods);
```

Given that the code runs OK (although only tested by me), I think it's OK! But no harm in adding an explicit `slice`?

Reply:
ah yup - I remember now. 

Should be OK, but also no harm in adding the slice